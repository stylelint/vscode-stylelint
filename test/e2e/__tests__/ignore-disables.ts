import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('"stylelint.ignoreDisables" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should ignore disable directives when enabled', async () => {
		const document = await openDocument('ignore-disables/test.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'color-hex-length',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-length',
				message: 'Expected "#fff" to be "#ffffff" (color-hex-length)',
				range: [3, 14, 3, 18],
				severity: 'error',
			},
		]);
	});
});
