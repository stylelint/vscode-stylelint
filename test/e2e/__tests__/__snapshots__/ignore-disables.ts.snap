// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.ignoreDisables" setting should ignore disable directives when enabled 1`] = `
[
  {
    "code": "color-hex-length",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/color-hex-length",
    },
    "message": "Expected "#fff" to be "#ffffff" (color-hex-length)",
    "range": {
      "end": {
        "character": 18,
        "line": 2,
      },
      "start": {
        "character": 14,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "function-name-case",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/function-name-case",
    },
    "message": "Expected "RGB" to be "rgb" (function-name-case)",
    "range": {
      "end": {
        "character": 23,
        "line": 2,
      },
      "start": {
        "character": 20,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
